### PR TITLE
Handle output path and configuration in .sqlproj

### PR DESCRIPTION
**Pull request by kisantia**
This PR fixes https://github.com/microsoft/azuredatastudio/issues/19918 and fixes https://github.com/microsoft/azuredatastudio/issues/20027. There was a previous PR that got reverted for handling output path because it didn't handle multiple OutputPaths in the .sqlproj correctly and take configuration into consideration. I made sure to do more thorough testing of both legacy and sdk-style projects this time and add more tests.

It looks like a lot of lines, but it's about 50 new lines for .sqlproj parsing, then tests and baseline files for those tests :)

### CODE REVIEW

File: extensions/sql-database-projects/src/common/constants.ts [hunk 1]
```diff
@@ -494,6 +499,11 @@ export const RoundTripSqlDbPresentCondition = '\'$(NetCoreBuild)\' != \'true\' A
 export const RoundTripSqlDbNotPresentCondition = '\'$(NetCoreBuild)\' != \'true\' AND \'$(SQLDBExtensionsRefPath)\' == \'\'';
 export const DacpacRootPath = '$(DacPacRootPath)';
 export const ProjJsonToClean = '$(BaseIntermediateOutputPath)\\project.assets.json';
+export const EmptyConfigurationCondition = '\'$(Configuration)\' == \'\'';
+export const EmptyPlatformCondition = '\'$(Platform)\' == \'\'';
+export function ConfigurationPlatformCondition(configuration: string, platform: string) { return `'$(Configuration)|$(Platform)' == '${configuration}|${platform}'`; }
```
Comment (kisantia) [inline extensions/sql-database-projects/src/common/constants.ts:504]: When checking the PropertyGroup condition when looking for OutputPath, only the conditions that are in legacy .sqlproj by default when creating a project in VS and ADS are handled for now.

Comment (SakshiS-harma) [reply extensions/sql-database-projects/src/common/constants.ts:504]: Just checking if you created an issue for the remaining part of the issue?

Comment (kisantia) [reply extensions/sql-database-projects/src/common/constants.ts:504]: just opened an issue for it now #20439 :)

**Review by zijchen — APPROVED** [review]
What if someone specifies OutputPath like `<OutputPath>out/$(Configuration)</OutputPath>`?

**Comment by kisantia** [comment]
> What if someone specifies OutputPath like `<OutputPath>out/$(Configuration)</OutputPath>`?

That won't work with the current xml parsing. The xml parsing code is already pretty messy in the sql projects extension, so I'm reluctant to keep adding to it. I opened https://github.com/microsoft/azuredatastudio/issues/20467 to track it while we're deciding if the parsing logic will move somewhere else.

**Comment by zijchen** [comment]
> > What if someone specifies OutputPath like `<OutputPath>out/$(Configuration)</OutputPath>`?
> 
> That won't work with the current xml parsing. The xml parsing code is already pretty messy in the sql projects extension, so I'm reluctant to keep adding to it. I opened #20467 to track it while we're deciding if the parsing logic will move somewhere else.

I agree that may require some heavy lifting. Do you know if there's a npm module similar to the [C# Microsoft.Build.Evaluation](https://docs.microsoft.com/en-us/dotnet/api/microsoft.build.evaluation.project?view=msbuild-17-netcore) that we can leverage for the project parsing?

**Comment by kisantia** [comment]
> > > What if someone specifies OutputPath like `<OutputPath>out/$(Configuration)</OutputPath>`?
> > 
> > 
> > That won't work with the current xml parsing. The xml parsing code is already pretty messy in the sql projects extension, so I'm reluctant to keep adding to it. I opened #20467 to track it while we're deciding if the parsing logic will move somewhere else.
> 
> I agree that may require some heavy lifting. Do you know if there's a npm module similar to the [C# Microsoft.Build.Evaluation](https://docs.microsoft.com/en-us/dotnet/api/microsoft.build.evaluation.project?view=msbuild-17-netcore) that we can leverage for the project parsing?

I've only found xml parsers, nothing with nice project specific apis like Microsoft.Build.Evaluation has :(

**Review by SakshiS-harma — APPROVED** [review]

**Pull request merged**